x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/TextEmbeddingConfig.java

```diff
@@ -59,22 +62,26 @@ private static ConstructingObjectParser<TextEmbeddingConfig, Void> createParser(
             TOKENIZATION
         );
         parser.declareString(ConstructingObjectParser.optionalConstructorArg(), RESULTS_FIELD);
+        parser.declareInt(ConstructingObjectParser.optionalConstructorArg(), EMBEDDING_SIZE);
         return parser;
     }
 
     private final VocabularyConfig vocabularyConfig;
     private final Tokenization tokenization;
     private final String resultsField;
+    private final Integer embeddingSize;
 
     public TextEmbeddingConfig(
         @Nullable VocabularyConfig vocabularyConfig,
         @Nullable Tokenization tokenization,
-        @Nullable String resultsField
+        @Nullable String resultsField,
+        @Nullable Integer embeddingSize
     ) {
         this.vocabularyConfig = Optional.ofNullable(vocabularyConfig)
             .orElse(new VocabularyConfig(InferenceIndexConstants.nativeDefinitionStore()));
         this.tokenization = tokenization == null ? Tokenization.createDefault() : tokenization;
         this.resultsField = resultsField;
+        this.embeddingSize = embeddingSize;
```

Comment:
Since this is serialized as strictly non-negative there should be a check in the public constructor that a negative number hasn't been supplied.

Reply:
👍 I've added that check to the ctor, it makes no sense for the size to be <= 0

commit: 1fbbdde6cae5df45815a8b4df267cfe602eed4a4